src/spikeinterface/core/waveforms_extractor_backwards_compatibility.py

```diff
@@ -350,6 +407,7 @@ def load_waveforms(
     if output == "SortingAnalyzer":
         return sorting_analyzer
     elif output in ("WaveformExtractor", "MockWaveformExtractor"):
+        sorting_analyzer.compute("random_spikes")
```

Comment:
Remove this and add `sorting_analyzer.extensions["random_spikes"] = ext` after the `SelectRandomSpikes` is instantiated in `_read_old_waveforms_extractor_binary`